sdk/protocol-manager-poc/src/interfaces/protocolPlugin.ts

```diff
@@ -394,18 +418,45 @@ export const createSparkPlugin: CreateProtocolPlugin<SparkPoolId> = (ctx: Protoc
         },
         getPosition: async (positionId: IPositionId): Promise<Position> => {
             throw new Error("Not implemented")
+        },
+        _validate: function (candidate: unknown): asserts candidate is SparkPoolId {
+            if (typeof candidate !== 'object' || candidate === null) {
+                throw new Error("Candidate is not an object");
+            }
+
+            if ('protocol' in candidate && 'emodeType' in candidate) {
+                const { protocol, emodeType } = candidate as { protocol: unknown, emodeType: unknown };
+
+                if (typeof protocol !== 'object' || protocol === null) {
+                    throw new Error("Protocol is not an object");
+                }
+
+                if ('name' in protocol && protocol.name !== ProtocolName.Spark) {
+                    throw new Error("Protocol is not Spark");
+                }
+
+                if (typeof emodeType !== 'string' || !isEModeType(emodeType)) {
+                    throw new Error("emodeType is invalid");
+                }
+            } else {
+                throw new Error("Candidate does not have the required properties 'protocol' and 'emodeType'");
+            }
```

Comment:
try using zod

Reply:
lol ;-)

Reply:
I was thinking about the same thing 😅 zod would be much easier to read

Reply:
I was thinking about using zod here but it's not really an issue and it's readable so should be fine